font/freetype: Enable bitmap glyphs for non-color faces

sin-ack opened the pull request:
This allows for crisp bitmap font rendering once again. Tested with Terminus (TTF), and at both 1x and 2x DPI the font renders perfectly (click for 1:1 size):

![Terminus (TTF), size 12, 96 DPI](https://github.com/user-attachments/assets/181ba561-ebe4-49df-aa41-68fc782f92b8)

<img alt="Terminus (TTF), size 12, 192 DPI" src="https://github.com/user-attachments/assets/d34804fe-4966-42e8-bf9e-bd10570ad443" width="384" height="50" />

sin-ack committed:
font/freetype: Rewrite monoToGrayscale algorithm

The original version had issues converting properly and caused broken
glyphs. This version tries to be as simple as possible in order to make
it easy to understand. I haven't measured the performance but in
practice this will only happen during the first render of the glyph
after a face change (i.e. during launch or when changing font size).

sin-ack committed:
font/freetype: Enable bitmap glyphs for non-color faces

This allows for crisp bitmap font rendering once again.

sin-ack commented:
@mitchellh Since you were the author of #1428, can you elaborate on what you mean by "issues with cell metrics and rasterization"?

mitchellh commented on src/font/face/freetype_convert.zig:
```diff
@@ -43,26 +43,14 @@ pub fn monoToGrayscale(alloc: Allocator, bm: Bitmap) Allocator.Error!Bitmap {
+            const byte_offset = row_offset + @divTrunc(x, 8);
+            const mask = @as(u8, 1) << @intCast(7 - (x % 8));
+            const bit: u8 = @intFromBool((bm.buffer[byte_offset] & mask) != 0);
+            buf[y * bm.width + x] = bit * 255;
```
On the same line of thinking I talked about in Discord, it'd be great if we added Terminus (license allowing) to `src/font/res` and add a unit test that failed before and passes now so that we don't regress this.

sin-ack replied:
Added Terminus TTF, and added a test that ensures that bitmap fonts keep being rendered crisply.

mitchellh commented on src/font/face/freetype.zig:
```diff
@@ -323,14 +322,6 @@ pub const Face = struct {
-            //
-            // This must be enabled for color faces though because those are
-            // often colored bitmaps, which we support.
-            .no_bitmap = !self.face.hasColor(),
```
Just noting I'm going to consider the implications of this a bit further. I don't remember the full history here.

sin-ack committed:
font/freetype: Add test for crisp bitmap font rendering

Now we can be certain that bitmap fonts stay crisp. :^)

sin-ack committed:
font/freetype: Downgrade pixfmt conversion log to debug

This is an expected occurrence with bitmap glyphs and causes unnecessary
spam when using the terminal with one.

mitchellh commented:
Okay, I've been running locally with this pulled in for a couple days, and I can't remember why that original comment was there. I think it's likely that Qwer's work on metrics just resolved that for us (or some other work along the way, since that line is _ancient_). Merging.
